docs/table/masking.rst

```diff
@@ -197,3 +210,100 @@ attribute.
   ---- ----
      1 1000
   1000    4
+
+.. _table-masked-4.0:
+
+Masking change in astropy 4.0
+=============================
+
+In astropy 4.0 a change was introduced in the behavior of |Table| that impacts the
+handling of masked columns.
+
+Prior to 4.0, in order to include one or more |MaskedColumn| columns in a table, it was
+required that *every* column be masked, even those with no missing or masked data.  This
+was holdover from the original implementation of |Table| that used a numpy structured
+array as the underlying container for the column data.  Since astropy 1.0 the |Table|
+object is simply an ordered dictionary of columns (:ref:`table_implementation_details`)
+and there is no requirement that column types be homogenous.
+
+Staring with 4.0, a |Table| can contain both |Column| and |MaskedColumn| columns, and
+by default the column type is determined solely by the data for each column.
+
+The details of this change are discussed in the sections below.
+
+.. Note::
+
+   For most applications we now recommend using the default |Table| behavior with
+   ``masked=False`` to allow heterogenous column types.
```

Comment:
This sounds like you recommend explicitly setting ``masked=False``, but the best is to just not specify ``masked=``, right?

Reply:
Ah, the wording was not ideal.  I'm changing to:
```
   For most applications, even those with masked column data, we now recommend using
   the default |Table| behavior which allows heterogenous column types.  This implies
   creating tables *without* specifying the ``masked`` keyword argument.
```